### PR TITLE
bug fix

### PR DESCRIPTION
**Pull request by shijl0925**

### CODE REVIEW

File: tpl/client.tpl [hunk 1]
```diff
@@ -145,7 +145,10 @@ func (c *Client) invoke(ctx context.Context, post bool, url string, payload inte
 
 	req.Header.Set("content-type", "application/x-www-form-urlencoded")
 
-	req.SetBasicAuth(c.username, c.password)
+    if len(c.username) != 0 && len(c.password) != 0 {
```
Comment (RidgeA) [inline tpl/client.tpl]: Adjust the indentation, please.

File: tpl/service.tpl [hunk 2]
```diff
@@ -97,7 +98,7 @@ type {{.RequestTypeName}} struct {
 	{{- if .Deprecated}}
 	// Deprecated since {{.DeprecatedSince.String}}
 	{{- end }}
-	{{.ParamName}} *string {{tick}}json:"{{.Key}},omitempty" url:"{{.Key}},omitempty"{{tick}}
+	{{.ParamName}} string {{- if $post }} {{tick}}json:"{{.Key}}{{ if not .Required}},omitempty{{ end }}"{{tick}} {{- else}} {{tick}}url:"{{.Key}}{{ if not .Required}},omitempty{{ end }}"{{tick}} {{- end}}
```
Comment (RidgeA) [inline tpl/service.tpl]: I consider this a breaking change, as the new version would generate a different output for the same input. 
Why do you want to change the field type?

**Review by RidgeA — CHANGES_REQUESTED** [review]

File: tpl/client.tpl (hunk 1 above)
Comment (shijl0925) [reply tpl/client.tpl]: Done!

File: tpl/service.tpl (hunk 2 above)
Comment (shijl0925) [reply tpl/service.tpl]: the change of the field type has been reverted.

**Review by RidgeA — APPROVED** [review]

**Comment by RidgeA** [comment]
Thanks @shijl0925

**Pull request merged**